src/mds/inode_backtrace.cc

```diff
@@ -140,7 +141,8 @@ int inode_backtrace_t::compare(const inode_backtrace_t& other,
        */
       break;
     }
-    if (ancestors[i].dirino != other.ancestors[i].dirino) {
+    if (ancestors[i].dirino != other.ancestors[i].dirino ||
+        ancestors[i].dname != other.ancestors[i].dname) {
```

Comment:
Okay, but do we really care about that when this is in use? We already update backtrackes on a local rename and if we don't have the object we're reconstructing the fragments anyway. What do we get by not considering them equivalent when the directory inodes still match, besides potentially extra IO?